fix: Addressed issues related to fetching different informations in private channels.

Spiral-Memory opened the pull request:
# Brief Title
Resolved issue with fetching messages, member, channel info and roles in private channels.

## Acceptance Criteria fulfillment

- [x] Identified the bug causing inability to fetch messages in private channels. Found that the API `channels.info` was unable to retrieve private channel information. Replaced it with `rooms.info` as per Rocket Chat REST API documentation. Introduced a state variable `isChannelPrivate` which will change based on `t` property of channel information to determine which API to use for fetching messages, members, and roles based on channel privacy.

- [x] Utilized `groups.*` API instead of `channel.*`  as per the documentation when fetching messages, members, and roles to address the limitations of the former in handling private channel information.

- [x] Adjusted frontend calls accordingly to accommodate the updated API. No breaking changes introduced, as `isChannelPrivate` defaults to false.

Fixes #423 

## Video/Screenshots


https://github.com/RocketChat/EmbeddedChat/assets/78961432/abf8409d-f228-4ad2-90c3-1250869dbd64

Note: In the video, zishan.barun@gmail.com is the authorized user and ahmadzishan7277@gmail.com is the unauthorized user. This is to demonstrate the functionality of the app.

sidmohanty11 commented on packages/react/src/components/ChatHeader/ChatHeader.js:
```diff
@@ -34,6 +34,8 @@ const ChatHeader = ({
   const setShowChannelinfo = useChannelStore(
     (state) => state.setShowChannelinfo
   );
+  const isChannelPrivate = useChannelStore((state)=>state.isChannelPrivate);
```
can you please run prettier?

Spiral-Memory replied:
Yaa sure, just a min

Spiral-Memory replied:
@sidmohanty11 , Done

sidmohanty11 reviewed (commented):
LGTM! Thanks @Spiral-Memory ! Just left a minor comment

Spiral-Memory commented:
@sidmohanty11 , Thank you so much for your appreciation 🚀, it means a lot. I've resolved the formatting issue as you asked. You can review it and approve / merge the changes if everything looks good.

Spiral-Memory commented:
@sidmohanty11 , @abhinavkrin , can either of you please merge this pull request? It has already been reviewed and i have fixed the formatting issue. I have another pull request, #421, which I have converted into a draft, as that can only be finalized after these changes are reflected. I will work on that after this, as it is dependent on this PR.

abhinavkrin commented on packages/api/src/EmbeddedChatApi.ts:
```diff
@@ -435,13 +435,14 @@ export default class EmbeddedChatApi {
    * @returns messages
    */
-  async getMessages(anonymousMode = false, options: {
+  async getMessages(anonymousMode = false, isChannelPrivate = false, options: {
```
This is a breaking change as you have changed the order of parameters. Add the new parameter at the last.

Spiral-Memory replied:
@abhinavkrin , Awesome catch It might break the react native app, Just a min, let me fix it.

Spiral-Memory commented:
@abhinavkrin , Fixed as per your suggestion.
